### PR TITLE
core remove package pycpuinfo

### PR DESCRIPTION
**Pull request by tensor-tang**
删除对python包py-cpuinfo的依赖
1. 减轻框架对第三方python的依赖。
2. 该包在MAC上判断AVX flag不太稳定。

### CODE REVIEW

**Comment by tensor-tang** [comment]
@tianshuo78520a Please help to review this PR

**Review by junjun315 — APPROVED** [review]
checked： `python/requirements.txt`

File: python/paddle/fluid/core.py [hunk 1]
```diff
@@ -57,27 +56,112 @@
...
+                    'sysctl machdep.cpu.leaf7_features | grep -i avx').read(
+                    ) != ''
+            except Exception as e:
+                sys.stderr.write(
+                    'Can not get the AVX flag from machdep.cpu.leaf7_features.\n'
+                    'The original error is: %s\n' %
+                    cpt.get_exception_message(e))
+        return has_avx
+    elif sysstr == 'windows':
+        import ctypes
```
Comment (JiabinYang) [inline python/paddle/fluid/core.py:96]: will this package be an dependency?

**Review by JiabinYang — APPROVED** [review]
LGTM

File: python/paddle/fluid/core.py (hunk 1 above)
Comment (tensor-tang) [reply python/paddle/fluid/core.py:96]: It seems in `The Python Standard Library` https://docs.python.org/3.6/library/ctypes.html

File: python/paddle/fluid/core.py [hunk 2]
```diff
@@ -57,27 +56,112 @@
...
+        ONE_PAGE = ctypes.c_size_t(0x1000)
 
-if has_avx:
+        def asm_func(code_str, restype=ctypes.c_uint32, argtypes=()):
```
Comment (wopeizl) [inline python/paddle/fluid/core.py:99]: not sure whether this would violate the DAP on windows

Comment (tensor-tang) [reply python/paddle/fluid/core.py:99]: It should be fine ctypes calls the windll, and protect the memory https://docs.microsoft.com/en-us/windows/win32/api/memoryapi/nf-memoryapi-virtualprotect

**Pull request merged**